plugins/nextcloud/nextcloud_

```diff
@@ -201,8 +215,10 @@ EOM
 esac
 
 # Get JSON data
-[ "$(test_https)" ] && DOMAIN="https://$DOMAIN" || DOMAIN="http://$DOMAIN"
-JSONSTATS=$(curl -s -f -m 2 -u "$USERNAME:$PASSWORD" "$DOMAIN$API_PATH" | sed 's/\\/\\\\/g' | jq -cr ".ocs.data")
+JSONSTATS=$(
+	fetch_url -u "$USERNAME:$PASSWORD" "https://${DOMAIN}${API_PATH}" | sed 's/\\/\\\\/g' | jq -cr ".ocs.data" 2>&1 \
```

Comment:
How about `{ fetch_url ... || fetch_url ...; } | sed ...` instead?

Reply:
The detection is a bit dodgy, and depends on whether we are able to parse JSON out of the received data, so we need to go all the way through `sed` and `jq` before we know if we've received what we want.

Reply:
I think, `curl --fail` (as you used above) should already take care for the proper detection of the API's availability. Or maybe you should add (`--location`?) in order to handle redirects (which could look like success, but have an empty body)?

Anyway: feel free to keep it, if you feel like.

Reply:
Alright, I bit that bullet. I was about to explain why it was done like this, and how it could be done better. Then I figured it would be best to actually do it. (:+1: 

So I added a `scheme` configuration variable (defaults to `https` if not set), and that removes all the uncertainty about what to hit, and the weird duplicates.

The CAVEAT is that it's a breaking change for servers where Nextcloud is only available over HTTP. It would have previously been autodetected, while it will now be required for `scheme http` to be present in the config for the plugin.